DuckDuckGo/Tab/Model/SystemInfo.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  SystemInfo.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import Common
+
+#if !APPSTORE
+
+final class SystemInfo {
+
+    static func pixelParameters(appVersion: AppVersion = AppVersion.shared) -> [String: String] {
+        let availableMemoryPercent = Self.getAvailableMemoryPercent()
+        let availableDiskSpacePercent = Self.getAvailableDiskSpacePercent()
+        return [
+           "available_memory": String(format: "%.2f", availableMemoryPercent),
+           "available_diskspace": String(format: "%.2f", availableDiskSpacePercent),
+           "os_version": appVersion.osVersion,
+        ]
+    }
+
+    static func getAvailableMemoryPercent() -> Int {
+        let process = Process()
+        process.executableURL = URL(fileURLWithPath: "/usr/bin/memory_pressure")
+
+        let pipe = Pipe()
+        process.standardOutput = pipe
+
+        do {
+            try process.run()
+            process.waitUntilExit()
```

Comment:
This runs on main thread now. It may become especially problematic if multiple crash pixels are being sent in a quick succession. Could you convert this to an asynchronous function and send pixel in the navigation delegate inside a Task?

Here's an inspiration: https://github.com/duckduckgo/macos-symbolication/blob/main/Sources/symbolicate/Extensions/Process%2BWhich.swift (it uses a different mechanism of capturing output, but we could stick to using `waitUntilExit` here 👍)

Reply:
Ah, good catch, thanks.